# Quality Check - Reconstructed Municipal Data

In [1]:
import pandas as pd

import pathlib
import os

from tqdm import tqdm

In [12]:
raw_path = pathlib.Path().resolve().parent.parent / "Dropbox" / "2019 MV Data by Town" / "Vehicles_2022" / "Compiled"

# Load data

In [63]:
recompiled_data = pd.read_csv(raw_path / "2019-21_data_compiled_RN_092923.csv", chunksize = 1000)

In [40]:
original_data = pd.read_csv(raw_path / "2019-21 data compiled.csv", chunksize = 1000)

# Checks

## Check the same files are used

In [36]:
recompiled_data_sources = []
recompiled_data_len = 0

Get all the unique file sources in the recompiled data

In [37]:
i = 0

for chunk in recompiled_data:
    if i %1000 == 0:
        print(f"Currently on chunk {i}")
        
    recompiled_data_len += len(chunk)
    
    sources = list(chunk["record_from"].unique())
    sources_new = [item for item in sources if not item in recompiled_data_sources]
    for item in sources_new:
        recompiled_data_sources.append(item)
        
    i +=1

Currently on chunk 0
Currently on chunk 1000
Currently on chunk 2000
Currently on chunk 3000
Currently on chunk 4000
Currently on chunk 5000


Get all the unique file sources in the original data

In [23]:
original_data_sources = []
original_data_len = 0

In [41]:
i = 0

for chunk in original_data:
    if i %1000 == 0:
        print(f"Currently on chunk {i}")
        
    original_data_len += len(chunk)
    
    sources = list(chunk["record_from"].unique())
    sources_new = [item for item in sources if not item in original_data_sources]
    for item in sources_new:
        original_data_sources.append(item)
        
    i +=1

Currently on chunk 0
Currently on chunk 1000
Currently on chunk 2000
Currently on chunk 3000
Currently on chunk 4000
Currently on chunk 5000


Compare them

In [43]:
diff1 = [item for item in original_data_sources if not item in recompiled_data_sources]
diff2 = [item for item in recompiled_data_sources if not item in original_data_sources]

diff1

['77_Manchester_MVData_2019.csv',
 '077_Manchester_MV_21.xls',
 '122_Salisbury_MV_21.xlsx',
 '016_Bridgewater_MV_21.xlsx']

This makes sense - these are the files that got edited. Manchester 2019 and 2021, Salisbury, and Bridgewater

In [46]:
diff2 = [item for item in diff2 if not "ALTERED" in item]

In [47]:
diff2

['010_Bethlehem_MV_21.xlsx',
 '020_Burlington_MV_21.xlsx',
 '024_Chaplin_MV_21.xlsx',
 '034_Danbury_MV_21.xlsx',
 '061_Haddam_MV_21.xlsx',
 '066_Harwinton_MV_21.xlsx',
 '079_Marlborough_MV_21.xlsx',
 '086_Montville_MV_21.xlsx',
 '090_New_Canaan_MV_21.xlsx',
 '131_Southington_MV_21.xlsx',
 '147_Voluntown_MV_21.xlsx',
 '163_Windham_MV_21.xlsx']

Why weren't these items in the original data sources?... They appear to have been skipped or added later.

In [65]:
investigate_diff2 = pd.DataFrame([])
i = 0

for chunk in recompiled_data:
    if i %1000 == 0:
        print(f"Currently on chunk {i}")
        
    required = chunk[chunk["record_from"].isin(diff2)]
        
    excerpt_head = required.head(10)
    excerpt_tail = required.tail(10)
    
    excerpt = pd.concat([excerpt_head, excerpt_tail])
    
    investigate_diff2 = pd.concat([investigate_diff2, excerpt])
        
    i +=1

Currently on chunk 0
Currently on chunk 1000
Currently on chunk 2000
Currently on chunk 3000
Currently on chunk 4000
Currently on chunk 5000


In [74]:
heads = []
for item in diff2:
    df = investigate_diff2[investigate_diff2["record_from"] == item]
    heads.append(df.head(10))

for df in heads:
    display(df)

Unnamed: 0               record_from  \
95937       95937  010_Bethlehem_MV_21.xlsx   
95938       95938  010_Bethlehem_MV_21.xlsx   
95939       95939  010_Bethlehem_MV_21.xlsx   
95940       95940  010_Bethlehem_MV_21.xlsx   
95941       95941  010_Bethlehem_MV_21.xlsx   
95942       95942  010_Bethlehem_MV_21.xlsx   
95943       95943  010_Bethlehem_MV_21.xlsx   
95944       95944  010_Bethlehem_MV_21.xlsx   
95945       95945  010_Bethlehem_MV_21.xlsx   
95946       95946  010_Bethlehem_MV_21.xlsx   

                                           name  \
95937  37-39 CITY HILL STREET LLC                 
95938  37-39 CITY HILL STREET LLC                 
95939  A & B HARD FLOOR LLC                       
95940  A MIM S CORP                               
95941  A MIM S CORP                               
95942  A MIMS CORP                                
95943  A MIMS CORP                                
95944  A MIMS CORP                                
95945  A MIMS CORP                                
95946  ABBEY OF REGINA LAUDIS                     

                               street                            city state  \
95937  15 W SHORE DR                   BETHLEHEM                         CT   
95938  15 W SHORE DR                   BETHLEHEM                         CT   
95939  PO BOX 1387                     WATERBURY                         CT   
95940  218 GUILDS HOLLOW RD            BETHLEHEM                         CT   
95941  218 GUILDS HOLLOW RD            BETHLEHEM                         CT   
95942  218 GUILDS HOLLOW RD            BETHLEHEM                         CT   
95943  218 GUILDS HOLLOW RD            BETHLEHEM                         CT   
95944  218 GUILDS HOLLOW RD            BETHLEHEM                         CT   
95945  218 GUILDS HOLLOW RD            BETHLEHEM                         CT   
95946  273 FLANDERS RD                 BETHLEHEM                         CT   

        zip vehicle_year vehicle_make vehicle_model vehicle_class  \
95937  6751       2021.0        CHEVR      SILVERAD           2.0   
95938  6751       2022.0        LOOK       STLC              10.0   
95939  6721       2019.0        GMC        SIERRA K           3.0   
95940  6751       2000.0        MACK       CX600              2.0   
95941  6751       2012.0        LANDO      T L               10.0   
95942  6751       2013.0        PTRB       CONVENTI           2.0   
95943  6751       1999.0        FONTA      303NDMRV          10.0   
95944  6751       1995.0        LANDO      317               10.0   
95945  6751       2004.0        INTER      4000 SER           2.0   
95946  6751       2005.0        HONDA      ACCORD L           1.0   

              vehicle_id                    lease_street  UID  \
95937  1GC4YUEY1MF128671  15 WEST SHORE DR                NaN   
95938  53BLTEA10NP023562  15 WEST SHORE DR                NaN   
95939  1GT12SEY7KF230918  21 SKY MEADOW RD                NaN   
95940  1M1AE07Y5YW003647                                  NaN   
95941  1LH440VHXC1018832                                  NaN   
95942  1XPSD79X2DD172162                                  NaN   
95943  4LF4G4825X3508389                                  NaN   
95944  1LH317VJ9S1007514                                  NaN   
95945  1HTMMAAM14H667912                                  NaN   
95946  1HGCM56495A149907                                  NaN   

                           lease_city lease_state lease_zip  
95937  BETHLEHEM                               CT     06751  
95938  BETHLEHEM                               CT     06751  
95939  BETHLEHEM                               CT     06751  
95940                                                        
95941                                                        
95942                                                        
95943                                                        
95944                                                        
95945                  

Unnamed: 0                record_from  \
318203      318203  020_Burlington_MV_21.xlsx   
318204      318204  020_Burlington_MV_21.xlsx   
318205      318205  020_Burlington_MV_21.xlsx   
318206      318206  020_Burlington_MV_21.xlsx   
318207      318207  020_Burlington_MV_21.xlsx   
318208      318208  020_Burlington_MV_21.xlsx   
318209      318209  020_Burlington_MV_21.xlsx   
318210      318210  020_Burlington_MV_21.xlsx   
318211      318211  020_Burlington_MV_21.xlsx   
318212      318212  020_Burlington_MV_21.xlsx   

                                            name  \
318203  A-1 ENTERPRISES LLC                        
318204  A-1 ENTERPRISES LLC                        
318205  ABALAN DOUGLAS L                           
318206  ABALAN DOUGLAS L                           
318207  ABALAN RINETTE R                           
318208  ABASCAL RICHARD                            
318209  ABASCAL RICHARD                            
318210  ABDELREHIM MOHAMED E                       
318211  ABDELREHIM MOHAMED E                       
318212  ABDELREHIM MOHAMED E                       

                                street                            city state  \
318203  8 MARY ROAD                     BURLINGTON                        CT   
318204  8 MARY ROAD                     BURLINGTON                        CT   
318205  71 MILFORD ST                   BURLINGTON                        CT   
318206  71 MILFORD ST                   BURLINGTON                        CT   
318207  71 MILFORD ST                   BURLINGTON                        CT   
318208  68 VENICE DR                    BURLINGTON                        CT   
318209  68 VENICE DR                    BURLINGTON                        CT   
318210  59 BELDEN RD                    BURLINGTON                        CT   
318211  59 BELDEN RD                    BURLINGTON                        CT   
318212  59 BELDEN RD                    BURLINGTON                        CT   

         zip vehicle_year vehicle_make vehicle_model vehicle_class  \
318203  6013       2021.0        FORD       F150 SUP             3   
318204  6013       2019.0        CHEVR      SILVERAD             2   
318205  6013       2011.0        DODGE      GRAND CR             1   
318206  6013       2004.0        DODGE      RAM 1500             1   
318207  6013       2001.0        HONDA      CR-V EX              1   
318208  6013       2013.0        TOYOT      TACOMA A             3   
318209  6013       2011.0        HARLE      FLSTN               12   
318210  6013       2005.0        HONDA      PILOT EX             1   
318211  6013       2013.0        TOYOT      CAMRY L/             1   
318212  6013       2012.0        HONDA      PILOT EX             1   

               vehicle_id                    lease_street  UID  \
318203  1FTFW1E55MKD52858                                  NaN   
318204  1HTKHPVK5KH406240  8 MARY RD                       NaN   
318205  2D4RN5DGXBR683379                                  NaN   
318206  1D7HU18D84S564872                                  NaN   
318207  JHLRD18661C024901                                  NaN   
318208  5TFUU4EN9DX054329                                  NaN   
318209  1HD1JD517BB017457                                  NaN   
318210  2HKYF18425H570235  59 BELDEN ROAD                  NaN   
318211  4T1BF1FK5DU660406                                  NaN   
318212  5FNYF4H55CB053260                                  NaN   

                            lease_city lease_state lease_zip  
318203                                                        
318204  BURLINGTON                              CT     06013  
318205                                                        
318206                                                        
318207                                                        
318208                                                        
318209                                                        
318210  BURL

Unnamed: 0             record_from                      name  \
345894      345894  024_Chaplin_MV_21.xlsx  ABARZUA DAGGETT ASCHLY C   
345895      345895  024_Chaplin_MV_21.xlsx  ABARZUA DAGGETT ASCHLY C   
345896      345896  024_Chaplin_MV_21.xlsx             ABELIN DAWN L   
345897      345897  024_Chaplin_MV_21.xlsx          ACAR LEASING LTD   
345898      345898  024_Chaplin_MV_21.xlsx          ACAR LEASING LTD   
345899      345899  024_Chaplin_MV_21.xlsx          ACAR LEASING LTD   
345900      345900  024_Chaplin_MV_21.xlsx          ACAR LEASING LTD   
345901      345901  024_Chaplin_MV_21.xlsx          ACAR LEASING LTD   
345902      345902  024_Chaplin_MV_21.xlsx          ACAR LEASING LTD   
345903      345903  024_Chaplin_MV_21.xlsx            ADAMS ALEXIS E   

                      street       city state    zip vehicle_year  \
345894          26 BEDLAM RD    CHAPLIN    CT   6235       2003.0   
345895          26 BEDLAM RD    CHAPLIN    CT   6235       2014.0   
345896        150 CHEWINK RD    CHAPLIN    CT   6235       2009.0   
345897   4001 EMBARCADERO DR  ARLINGTON    TX  76014       2021.0   
345898   4001 EMBARCADERO DR  ARLINGTON    TX  76014       2018.0   
345899   4001 EMBARCADERO DR  ARLINGTON    TX  76014       2019.0   
345900   4001 EMBARCADERO DR  ARLINGTON    TX  76014       2021.0   
345901   4001 EMBARCADERO DR  ARLINGTON    TX  76014       2020.0   
345902   4001 EMBARCADERO DR  ARLINGTON    TX  76014       2019.0   
345903  20 CAREFREE LN APT 1    CHAPLIN    CT   6235       2018.0   

       vehicle_make vehicle_model vehicle_class         vehicle_id  \
345894        HONDA      VT750DCB            12  JH2RC44503M708944   
345895        VOLKS      PASSAT S             1  1VWBN7A38EC063894   
345896        HONDA       CR-V EX             1  3CZRE48539G703675   
345897        BUICK      ENCORE P             1  KL4CJESM2MB361970   
345898        BUICK      ENCORE P             1  KL4CJESB7JB694739   
345899        CHEVR      TRAX 1LT             1  KL7CJPSBXKB951800   
345900          GMC       TERRAIN             1  3GKALVEV6ML351991   
345901        CHEVR      SILVERAD             3  1GCRYEED2LZ369807   
345902          GMC      SIERRA L             3  2GTV2LEC7K1105839   
345903        MITSU      OUTLANDE             1  JA4AP3AWXJU023925   

       lease_street  UID lease_city lease_state lease_zip  
345894          NaN  NaN        NaN         NaN       NaN  
345895          NaN  NaN        NaN         NaN       NaN  
345896          NaN  NaN        NaN         NaN       NaN  
345897          NaN  NaN        NaN         NaN       NaN  
345898          NaN  NaN        NaN         NaN       NaN  
345899          NaN  NaN        NaN         NaN       NaN  
345900          NaN  NaN        NaN         NaN       NaN  
345901          NaN  NaN        NaN         NaN       NaN  
345902          NaN  NaN        NaN         NaN       NaN  
345903          NaN  NaN        NaN         NaN       NaN

Unnamed: 0             record_from  \
421301      421301  034_Danbury_MV_21.xlsx   
421302      421302  034_Danbury_MV_21.xlsx   
421303      421303  034_Danbury_MV_21.xlsx   
421304      421304  034_Danbury_MV_21.xlsx   
421305      421305  034_Danbury_MV_21.xlsx   
421306      421306  034_Danbury_MV_21.xlsx   
421307      421307  034_Danbury_MV_21.xlsx   
421308      421308  034_Danbury_MV_21.xlsx   
421309      421309  034_Danbury_MV_21.xlsx   
421310      421310  034_Danbury_MV_21.xlsx   

                                            name  \
421301  29 FEDERAL ROAD LLC                        
421302  2J S LLC                                   
421303  2M GENERAL SERVICES LLC                    
421304  3M GENERAL SERVICES LLC                    
421305  4 SEASONSLLC                               
421306  4 SEASONSLLC                               
421307  404 CAR LINE LLC                           
421308  46 SL LLC                                  
421309  85 MILL PLAIN ROAD LLC                     
421310  A & C ROOFING AND SIDING LLC               

                                street                            city state  \
421301  64 TRIANGLE ST.                 DANBURY                           CT   
421302  12 STARR AVE 14 STARR AVE       DANBURY                           CT   
421303  12 TRIANGLE ST UNIT 2           DANBURY                           CT   
421304  13 COTTAGE ST FL 2              DANBURY                           CT   
421305  50 NORTH STREET                 DANBURY                           CT   
421306  50 NORTH STREET                 DANBURY                           CT   
421307  89 WALNUT TRL                   DANBURY                           CT   
421308  16 HAYESTOWN RD UNIT 3104       DANBURY                           CT   
421309  85 MILL PLAIN RD                DANBURY                           CT   
421310  8 MALLORY ST APT 2              DANBURY                           CT   

         zip vehicle_year vehicle_make vehicle_model vehicle_class  \
421301  6810         2007        CHEVR      EXPRESS              3   
421302  6811         2014        HINO       HINO 268             2   
421303  6810         1998        FORD       ECONOLIN             3   
421304  6810         2017        FORD       FUSION S             3   
421305  6810         2021        CROSS      61435               11   
421306  6810         2019        CHEVR      EXPRESS              3   
421307  6811         1999        FORD       EXPEDITI             3   
421308  6811         2020        PORSC      MACAN                1   
421309  6811         2017        RAM        RAM CHAS            70   
421310  6810         2008        GMC        ACADIA S             1   

               vehicle_id lease_street  UID lease_city lease_state lease_zip  
421301  1GBGG29U171156648          NaN  NaN        NaN         NaN       NaN  
421302  5PVNJ8JT9E4S55479          NaN  NaN        NaN         NaN       NaN  
421303  1FTRE1429WHA50822          NaN  NaN        NaN         NaN       NaN  
421304  3FA6P0HD1HR195797          NaN  NaN        NaN         NaN       NaN  
421305  431FS1411M1000212          NaN  NaN        NaN         NaN       NaN  
421306  1GCWGAFG3K1255867          NaN  NaN        NaN         NaN       NaN  
421307  1FMPU18L3XLA05477          NaN  NaN        NaN         NaN       NaN  
421308  WP1AA2A57LLB09404          NaN  NaN        NaN         NaN       NaN  
421309  3C7WRNDL8HG625578          NaN  NaN        NaN         NaN       NaN  
421310  1GKEV23768J245536          NaN  NaN        NaN         NaN       NaN

Unnamed: 0            record_from  \
938991      938991  061_Haddam_MV_21.xlsx   
938992      938992  061_Haddam_MV_21.xlsx   
938993      938993  061_Haddam_MV_21.xlsx   
938994      938994  061_Haddam_MV_21.xlsx   
938995      938995  061_Haddam_MV_21.xlsx   
938996      938996  061_Haddam_MV_21.xlsx   
938997      938997  061_Haddam_MV_21.xlsx   
938998      938998  061_Haddam_MV_21.xlsx   
938999      938999  061_Haddam_MV_21.xlsx   
938991      938991  061_Haddam_MV_21.xlsx   

                                            name  \
938991  A & A PROPERTY MAINTENANCE LLC             
938992  ABARIENTOS ANTONIETA L                     
938993  ABARIENTOS CRISPIN                         
938994  ABARIENTOS CRISPIN                         
938995  ABBATELLO STEVEN G                         
938996  ABBATELLO STEVEN G                         
938997  ABBATELLO STEVEN G                         
938998  ABBOTT ARTHUR H JR                         
938999  ABBOTT ARTHUR H JR                         
938991  A & A PROPERTY MAINTENANCE LLC             

                                street                            city state  \
938991  77 OAK RIDGE DR UNIT 1          HADDAM                            CT   
938992  17 LARKSPUR DR                  HIGGANUM                          CT   
938993  17 LARKSPUR DR                  HIGGANUM                          CT   
938994  17 LARKSPUR DR                  HIGGANUM                          CT   
938995  5 MAPLE AVE WEST                HIGGANUM                          CT   
938996  5 MAPLE AVE WEST                HIGGANUM                          CT   
938997  5 MAPLE AVE WEST                HIGGANUM                          CT   
938998  623 CANDLEWOOD HILL RD          HIGGANUM                          CT   
938999  623 CANDLEWOOD HILL RD          HIGGANUM                          CT   
938991  77 OAK RIDGE DR UNIT 1          HADDAM                            CT   

           zip vehicle_year vehicle_make vehicle_model vehicle_class  \
938991  6438.0         2019        FORD       F550 SUP             2   
938992  6441.0         2005        ACURA      MDX                  1   
938993  6441.0         2011        AUDI       A4 2.0T              1   
938994  6441.0         2001        PORSC      911 CARR             1   
938995  6441.0         1985        CHEVR      K10                 25   
938996  6441.0         2011        VOLKS      JETTA TD             1   
938997  6441.0         2003        CHEVR      SILVERAD             1   
938998  6441.0         2009        SUBAR      OUTBACK              1   
938999  6441.0         1997        CHEVR      K1500                3   
938991  6438.0         2019        FORD       F550 SUP             2   

               vehicle_id lease_street  UID lease_city lease_state lease_zip  
938991  1FD0X5HT6KEF91010          NaN  NaN        NaN         NaN       NaN  
938992  2HNYD182X5H552606          NaN  NaN        NaN         NaN       NaN  
938993  WAUFFAFL9BA091621          NaN  NaN        NaN         NaN       NaN  
938994  WP0AA299X1S620343          NaN  NaN        NaN         NaN       NaN  
938995  2GCEK14H3F1149599          NaN  NaN        NaN         NaN       NaN  
938996  3VWPL7AJ2BM616673          NaN  NaN        NaN         NaN       NaN  
938997  1GCEC14X73Z353572          NaN  NaN        NaN         NaN       NaN  
938998  4S4BP61C497330465          NaN  NaN        NaN         NaN       NaN  
938999  1GCEK14MXVZ191674          NaN  NaN        NaN         NaN       NaN  
938991  1FD0X5HT6KEF91010          NaN  NaN        NaN         NaN       NaN

Unnamed: 0               record_from  \
1043778     1043778  066_Harwinton_MV_21.xlsx   
1043779     1043779  066_Harwinton_MV_21.xlsx   
1043780     1043780  066_Harwinton_MV_21.xlsx   
1043781     1043781  066_Harwinton_MV_21.xlsx   
1043782     1043782  066_Harwinton_MV_21.xlsx   
1043783     1043783  066_Harwinton_MV_21.xlsx   
1043784     1043784  066_Harwinton_MV_21.xlsx   
1043785     1043785  066_Harwinton_MV_21.xlsx   
1043786     1043786  066_Harwinton_MV_21.xlsx   
1043787     1043787  066_Harwinton_MV_21.xlsx   

                                             name  \
1043778  .SUPREME INDUSTRIES INC.                   
1043779  .SUPREME INDUSTRIES INC.                   
1043780  362 POST ROAD LLC                          
1043781  ABBAMONDI CYNTHIA J                        
1043782  ABBOTTS JEFFREY C                          
1043783  ABBOTTS JEFFREY C                          
1043784  ABBOTTS JEFFREY C                          
1043785  ABELING CHRISTOPHER L                      
1043786  ABLING-JOSEPHSON CORITA P                  
1043787  ABOU ARRAGE ANDRE                          

                                 street                            city state  \
1043778  216 LOWER BOGUE RD              HARWINTON                         CT   
1043779  216 LOWER BOGUE RD              HARWINTON                         CT   
1043780  216 BOGUE RD                    HARWINTON                         CT   
1043781  30 HITHER LANE                  HARWINTON                         CT   
1043782  95 WILDCAT HILL RD              HARWINTON                         CT   
1043783  95 WILDCAT HILL RD              HARWINTON                         CT   
1043784  95 WILDCAT HILL RD              HARWINTON                         CT   
1043785  576 MAIN ST                     TORRINGTON                        CT   
1043786  352 CLEARVIEW AVE               HARWINTON                         CT   
1043787  34 SAND HILL LN                 GLASTONBURY                       CT   

          zip vehicle_year vehicle_make vehicle_model vehicle_class  \
1043778  6791         2019        CHEVR      SILVERAD             1   
1043779  6791         2017        CHEVR      SILVERAD             3   
1043780  6791         2018        TIDEW      TP-22-36            11   
1043781  6791         2016        VOLVO      XC60 T5              1   
1043782  6791         1983        FIAT       PININFAR            25   
1043783  6791         2013        VOLKS      BEETLE T             1   
1043784  6791         2016        RAM        RAM TRUC             3   
1043785  6790         2011        HONDA      ELEMENT              1   
1043786  6791         2012        SUBAR      FORESTER             1   
1043787  6033         2017        HONDA      ACCORD T             1   

                vehicle_id lease_street  UID lease_city lease_state lease_zip  
1043778  1GCUYDED3KZ127370          NaN  NaN        NaN         NaN       NaN  
1043779  1GC1KVEG7HF169684          NaN  NaN        NaN         NaN       NaN  
1043780  45LBS2219J2100584          NaN  NaN        NaN         NaN       NaN  
1043781  YV4612RM7G2831290          NaN  NaN        NaN         NaN       NaN  
1043782  ZFRAS00B3D5503101          NaN  NaN        NaN         NaN       NaN  
1043783  3VW4A7AT1DM617780          NaN  NaN        NaN         NaN       NaN  
1043784  3C6JR7DTXGG302349          NaN  NaN        NaN         NaN       NaN  
1043785  5J6YH2H75BL004405          NaN  NaN        NaN         NaN       NaN  
1043786  JF2SHADC7CH451624          NaN  NaN        NaN         NaN       NaN  
1043787  1HGCR3F90HA007645          NaN  NaN        NaN         NaN       NaN

Unnamed: 0                 record_from name  \
1199237     1199237  079_Marlborough_MV_21.xlsx  NaN   
1199238     1199238  079_Marlborough_MV_21.xlsx  NaN   
1199239     1199239  079_Marlborough_MV_21.xlsx  NaN   
1199240     1199240  079_Marlborough_MV_21.xlsx  NaN   
1199241     1199241  079_Marlborough_MV_21.xlsx  NaN   
1199242     1199242  079_Marlborough_MV_21.xlsx  NaN   
1199243     1199243  079_Marlborough_MV_21.xlsx  NaN   
1199244     1199244  079_Marlborough_MV_21.xlsx  NaN   
1199245     1199245  079_Marlborough_MV_21.xlsx  NaN   
1199246     1199246  079_Marlborough_MV_21.xlsx  NaN   

                                 street                            city state  \
1199237  32 EAST HAMPTON RD              MARLBOROUGH                       CT   
1199238  PO BOX 453                      MARLBOROUGH                       CT   
1199239  PO BOX 453                      MARLBOROUGH                       CT   
1199240  PO BOX 453                      MARLBOROUGH                       CT   
1199241  2 AVALON LN                     MARLBOROUGH                       CT   
1199242  2 AVALON LN                     MARLBOROUGH                       CT   
1199243  2 AVALON LN                     MARLBOROUGH                       CT   
1199244  2 AVALON LN                     MARLBOROUGH                       CT   
1199245  20 CARRIAGE LN                  MARLBOROUGH                       CT   
1199246  20 APACHE LN                    MARLBOROUGH                       CT   

          zip vehicle_year vehicle_make vehicle_model vehicle_class  \
1199237  6447         2018        TOYOT      4RUNNER              1   
1199238  6447         1999        FORD       F550 SUP             2   
1199239  6447         2001        ECONO                          10   
1199240  6447         2016        CHEVR      SILVERAD             3   
1199241  6447         2018        PORSC      MACAN                1   
1199242  6447         2015        AUDI       A4 2.0T              1   
1199243  6447         2020        DUCAT      DIAVEL 1            12   
1199244  6447         2018        FORD       F150 RAP             1   
1199245  6447         2016        NISSA      ROGUE S/             1   
1199246  6447         2019        HONDA      CR-V EXL             1   

                vehicle_id                    lease_street      UID  \
1199237  JTEBU5JR9J5607957  32 EAST HAMPTON RD RT 66        50001.0   
1199238  1FDAF56F0XEA33195  5 SHERWOOD LANE                 50002.0   
1199239  42EDPFB2X11001464  5 SHERWOOD LANE                 50003.0   
1199240  1GC1KUEG4GF278550  5 SHERWOOD LANE                 50004.0   
1199241  WP1AA2A59JLB15167                                  50005.0   
1199242  WAUBFBFL2FN040366                                  50006.0   
1199243  ZDMGAHRW2LB003903                                  50007.0   
1199244  1FTFW1RG0JFE52897                                  50008.0   
1199245  KNMAT2MV8GP723352                                  50009.0   
1199246  5J6RW2H8XKA000252                                  50010.0   

                             lease_city lease_state lease_zip  
1199237  MARLBOROUGH                             CT     06447  
1199238  MARLBOROUGH                             CT     06447  
1199239  MARLBOROUGH                             CT     06447  
1199240  MARLBOROUGH                             CT     06447  
1199241                                                        
1199242                                                        
1199243                                                        
1199244                                                        
1199245                                                        
1199246

Unnamed: 0               record_from  \
1364806     1364806  086_Montville_MV_21.xlsx   
1364807     1364807  086_Montville_MV_21.xlsx   
1364808     1364808  086_Montville_MV_21.xlsx   
1364809     1364809  086_Montville_MV_21.xlsx   
1364810     1364810  086_Montville_MV_21.xlsx   
1364811     1364811  086_Montville_MV_21.xlsx   
1364812     1364812  086_Montville_MV_21.xlsx   
1364813     1364813  086_Montville_MV_21.xlsx   
1364814     1364814  086_Montville_MV_21.xlsx   
1364815     1364815  086_Montville_MV_21.xlsx   

                                             name  \
1364806  A & B EXCAVATING L.L.C.                    
1364807  A & B EXCAVATING LLC                       
1364808  A & B EXCAVATING LLC                       
1364809  A & B EXCAVATING LLC                       
1364810  A & B EXCAVATING LLC                       
1364811  A & B TREE SERVICE LLC                     
1364812  A & B TREE SERVICES LLC                    
1364813  A J CABRAL TRUCKING                        
1364814  AANENSEN KENNETH T                         
1364815  AANENSEN KENNETH T JR                      

                                 street                            city state  \
1364806  33 COVE RD                      UNCASVILLE                        CT   
1364807  33 COVE RD                      UNCASVILLE                        CT   
1364808  33 COVE RD                      UNCASVILLE                        CT   
1364809  33 COVE RD                      UNCASVILLE                        CT   
1364810  33 COVE RD                      UNCASVILLE                        CT   
1364811  PO BOX 335                      MONTVILLE                         CT   
1364812  PO BOX 335                      MONTVILLE                         CT   
1364813  12 BONVILLE DR                  UNCASVILLE                        CT   
1364814  8 ROBIN LN                      OAKDALE                           CT   
1364815  8 ROBIN LN                      OAKDALE                           CT   

          zip vehicle_year vehicle_make vehicle_model vehicle_class  \
1364806  6382         1988        INTER      C16                 10   
1364807  6382         2013        FORD       F550 SUP             2   
1364808  6382         1997        INTER      4000 SER             2   
1364809  6382         2005        FORD       F350 SUP             2   
1364810  6382         1998        MITSU      FE639                2   
1364811  6353         1996        GMC        TOPKICK              2   
1364812  6353         2012        INTER      4000 SER             2   
1364813  6382         1994        KENWO      CONSTRUC             2   
1364814  6370         2021        NISSA      MURANO P             1   
1364815  6370         2009        HONDA      PILOT EX             1   

                vehicle_id lease_street  UID lease_city lease_state lease_zip  
1364806  1ZFCF1622JB003085          NaN  NaN        NaN         NaN       NaN  
1364807  1FD0W5HT9DEA34887          NaN  NaN        NaN         NaN       NaN  
1364808  1HTSCAAM2VH424900          NaN  NaN        NaN         NaN       NaN  
1364809  1FDWX37P75EB40966          NaN  NaN        NaN         NaN       NaN  
1364810  JW6AAE1H5WL001462          NaN  NaN        NaN         NaN       NaN  
1364811  1GDL7H1P4TJ512163          NaN  NaN        NaN         NaN       NaN  
1364812  3HAMMAAM6CL549363          NaN  NaN        NaN         NaN       NaN  
1364813  2XKDD69X9RM635618          NaN  NaN        NaN         NaN       NaN  
1364814  5N1AZ2DS3MC136494          NaN  NaN        NaN         NaN       NaN  
1364815  5FNYF48509B031076          NaN  NaN        NaN         NaN       NaN

Unnamed: 0                record_from  \
1431120     1431120  090_New_Canaan_MV_21.xlsx   
1431121     1431121  090_New_Canaan_MV_21.xlsx   
1431122     1431122  090_New_Canaan_MV_21.xlsx   
1431123     1431123  090_New_Canaan_MV_21.xlsx   
1431124     1431124  090_New_Canaan_MV_21.xlsx   
1431125     1431125  090_New_Canaan_MV_21.xlsx   
1431126     1431126  090_New_Canaan_MV_21.xlsx   
1431127     1431127  090_New_Canaan_MV_21.xlsx   
1431128     1431128  090_New_Canaan_MV_21.xlsx   
1431129     1431129  090_New_Canaan_MV_21.xlsx   

                                             name  \
1431120  1330 EAST PUTNAM LLC                       
1431121  1ST RIDE & GO CORP                         
1431122  909 WEST ROAD REVOCABLE LIVING TRUST       
1431123  A & N PLUMBING AND HEATING LLC             
1431124  A & N PLUMBING AND HEATING LLC             
1431125  A M SANTELLA COMPANY INC                   
1431126  A M SANTELLA COMPANY INC                   
1431127  A M SANTELLA COMPANY INC                   
1431128  A SILLO DEVELOPMENT LLC                    
1431129  A SILLO DEVELOPMENT LLC                    

                                 street                            city state  \
1431120  PO BOX 811                      NEW CANAAN                        CT   
1431121  58 PINE ST                      NEW CANAAN                        CT   
1431122  909 WEST RD                     NEW CANAAN                        CT   
1431123  22 DOWN RIVER RD                NEW CANAAN                        CT   
1431124  22 DOWN RIVER RD                NEW CANAAN                        CT   
1431125  635 CHEESEPRING RD              NEW CANAAN                        CT   
1431126  635 CHEESEPRING RD              NEW CANAAN                        CT   
1431127  635 CHEESEPRING RD              NEW CANAAN                        CT   
1431128  691 OLD STAMFORD ROAD           NEW CANAAN                        CT   
1431129  691 OLD STAMFORD ROAD           NEW CANAAN                        CT   

            zip vehicle_year vehicle_make vehicle_model vehicle_class  \
1431120  6840.0       2019.0        CHEVR      EQUINOX            1.0   
1431121  6840.0       2016.0        FORD       TRANSIT            8.0   
1431122  6840.0       1996.0        JAGUA      XJR               25.0   
1431123  6840.0       2007.0        FORD       ECONOLIN           3.0   
1431124  6840.0       2018.0        RAM        PROMASTE           3.0   
1431125  6840.0       2015.0        RAM        RAM TRUC           3.0   
1431126  6840.0       2007.0        ISUZU      NPR                2.0   
1431127  6840.0       2006.0        CAM        6CAM 18           10.0   
1431128  6840.0       2012.0        CHEVR      SILVERAD           2.0   
1431129  6840.0       2020.0        CHEVR      SILVERAD           3.0   

                vehicle_id                    lease_street    UID  \
1431120  3GNAXVEX0KS674784  261 ELM ST                      50001   
1431121  1FBZX2CM2GKA71622                                  50002   
1431122  SAJPX1143TC781554                                  50003   
1431123  1FTNE24L07DA36751                                  50004   
1431124  3C6TRVAG6JE110384                                  50005   
1431125  3C6MR5AJ2FG524435                                  50006   
1431126  JALB4W16277400371                                  50007   
1431127  5JPBU23296P015771                                  50008   
1431128  1GB3KZCG1CF113627  691 OLD STAMFORD RD             50009   
1431129  3GCPYFED0LG246956  691 OLD STAMFORD RD             50010   

                             lease_city lease_state lease_zip  
1431120  NEW CANAAN                              CT     06840  
1431121                                                        
1431122                                                        
1431123                                                        
1431124                                                        
1431125                           

Unnamed: 0                 record_from  \
2076515     2076515  131_Southington_MV_21.xlsx   
2076516     2076516  131_Southington_MV_21.xlsx   
2076517     2076517  131_Southington_MV_21.xlsx   
2076518     2076518  131_Southington_MV_21.xlsx   
2076519     2076519  131_Southington_MV_21.xlsx   
2076520     2076520  131_Southington_MV_21.xlsx   
2076521     2076521  131_Southington_MV_21.xlsx   
2076522     2076522  131_Southington_MV_21.xlsx   
2076523     2076523  131_Southington_MV_21.xlsx   
2076524     2076524  131_Southington_MV_21.xlsx   

                                        name             street          city  \
2076515                         18FORE18 INC  201 PATTONWOOD DR   SOUTHINGTON   
2076516  3 BROTHERS PROPERTY PRESERVATION SE      461C COOKE ST    FARMINGTON   
2076517                   A & A SURPLUS INC.     389 MARION AVE   PLANTSVILLE   
2076518                   A & A SURPLUS INC.     389 MARION AVE   PLANTSVILLE   
2076519                  A AND A SURPLUS INC     389 MARION AVE   PLANTSVILLE   
2076520          A AND P KRISHNA CORPORATION   151 QUEEN STREET   SOUTHINGTON   
2076521            A CUT ABOVE GREENCARE LLC   41 HIGHWOOD AVE.   SOUTHINGTON   
2076522                      A DUIE PYLE INC         PO BOX 564  WEST CHESTER   
2076523                      A DUIE PYLE INC         PO BOX 564  WEST CHESTER   
2076524                      A DUIE PYLE INC         PO BOX 564  WEST CHESTER   

        state      zip vehicle_year vehicle_make vehicle_model vehicle_class  \
2076515    CT   6489.0         2002        USCAR      USCL58SA            11   
2076516    CT   6032.0         2013         FORD      F350 SUP             3   
2076517    CT   6479.0         2008        TOYOT      TUNDRA D             3   
2076518    CT   6479.0         2012        FREIG      M2 106 M             2   
2076519    CT   6479.0         1999        INTER      4000 SER             2   
2076520    CT   6489.0         2015        MERCE      SPRINTER             3   
2076521    CT   6489.0         2017        FREED        6X10SA            10   
2076522    PA  19381.0         2012         FRHT          CA12             2   
2076523    PA  19381.0         2012         FRHT          CA12             2   
2076524    PA  19381.0         2012         FRHT          CA12             2   

                vehicle_id lease_street  UID lease_city lease_state lease_zip  
2076515  4X4TSE0112X053369          NaN  NaN        NaN         NaN       NaN  
2076516  1FTRF3B68DEA53496          NaN  NaN        NaN         NaN       NaN  
2076517  5TBBV54198S491700          NaN  NaN        NaN         NaN       NaN  
2076518  1FVACXDT5CHBV1762          NaN  NaN        NaN         NaN       NaN  
2076519  1HTSCAAM3XH688887          NaN  NaN        NaN         NaN       NaN  
2076520  WD3PF1CC9FP167427          NaN  NaN        NaN         NaN       NaN  
2076521  5WKBE101XH1045634          NaN  NaN        NaN         NaN       NaN  
2076522  1FUBGEDV9CLBK5195          NaN  NaN        NaN         NaN       NaN  
2076523  1FUBGEDV0CLBK5165          NaN  NaN        NaN         NaN       NaN  
2076524  1FUBGEDV0CLBK5196          NaN  NaN        NaN         NaN       NaN

Unnamed: 0               record_from  \
2375159     2375159  147_Voluntown_MV_21.xlsx   
2375160     2375160  147_Voluntown_MV_21.xlsx   
2375161     2375161  147_Voluntown_MV_21.xlsx   
2375162     2375162  147_Voluntown_MV_21.xlsx   
2375163     2375163  147_Voluntown_MV_21.xlsx   
2375164     2375164  147_Voluntown_MV_21.xlsx   
2375165     2375165  147_Voluntown_MV_21.xlsx   
2375166     2375166  147_Voluntown_MV_21.xlsx   
2375167     2375167  147_Voluntown_MV_21.xlsx   
2375168     2375168  147_Voluntown_MV_21.xlsx   

                                             name  \
2375159  A KAUSCH & SONS LLC                        
2375160  ACAR LEASING LTD                           
2375161  ACAR LEASING LTD                           
2375162  ACAR LEASING LTD                           
2375163  ACAR LEASING LTD                           
2375164  ACAR LEASING LTD                           
2375165  ACAR LEASING LTD                           
2375166  ACAR LEASING LTD                           
2375167  ACAR LEASING LTD                           
2375168  ADAMS ANGELA D                             

                                 street                            city state  \
2375159  15 BEACH VIEW ROAD EXT          VOLUNTOWN                         CT   
2375160  4001 EMBARCADERO DR             ARLINGTON                         TX   
2375161  4001 EMBARCADERO DR             ARLINGTON                         TX   
2375162  4001 EMBARCADERO DR             ARLINGTON                         TX   
2375163  4001 EMBARCADERO DR             ARLINGTON                         TX   
2375164  4001 EMBARCADERO DR             ARLINGTON                         TX   
2375165  4001 EMBARCADERO DR             ARLINGTON                         TX   
2375166  4001 EMBARCADERO DR             ARLINGTON                         TX   
2375167  4001 EMBARCADERO DR             ARLINGTON                         TX   
2375168  66 TEN ROD RD                   VOLUNTOWN                         CT   

           zip vehicle_year vehicle_make vehicle_model vehicle_class  \
2375159   6384         2006        CHEVR      EXPRESS              2   
2375160  76014         2019        GMC        ACADIA S             1   
2375161  76014         2019        CHEVR      EQUINOX              1   
2375162  76014         2019        CHEVR      SILVERAD             1   
2375163  76014         2020        CHEVR      SILVERAD             1   
2375164  76014         2020        CHEVR      SILVERAD             1   
2375165  76014         2021        CHEVR      SILVERAD             3   
2375166  76014         2021        GMC        YUKON DE             3   
2375167  76014         2021        GMC        ACADIA S             1   
2375168   6384         2014        MERCE      C300 4 M             1   

                vehicle_id lease_street  UID lease_city lease_state lease_zip  
2375159  1GBJG31U761268117          NaN  NaN        NaN         NaN       NaN  
2375160  1GKKNULS2KZ110890          NaN  NaN        NaN         NaN       NaN  
2375161  2GNAXUEV3K6194780          NaN  NaN        NaN         NaN       NaN  
2375162  1GCRYCEF6KZ368941          NaN  NaN        NaN         NaN       NaN  
2375163  1GCPYFED7LZ165405          NaN  NaN        NaN         NaN       NaN  
2375164  1GCRYBEH8LZ271768          NaN  NaN        NaN         NaN       NaN  
2375165  1GCRYBEH8MZ187550          NaN  NaN        NaN         NaN       NaN  
2375166  1GKS2DKL1MR127895          NaN  NaN        NaN         NaN       NaN  
2375167  1GKKNULS2MZ210281          NaN  NaN        NaN         NaN       NaN  
2375168  WDDGF8AB8EA949962          NaN  NaN        NaN         NaN       NaN

Unnamed: 0             record_from                     name  \
2691877     2691877  163_Windham_MV_21.xlsx  0 TO 60 MOTORSPORTS LLC   
2691878     2691878  163_Windham_MV_21.xlsx  A L FIRE PROTECTION LLC   
2691879     2691879  163_Windham_MV_21.xlsx   A+ PUMPS & SERVICE LLC   
2691880     2691880  163_Windham_MV_21.xlsx   A+ PUMPS & SERVICE LLC   
2691881     2691881  163_Windham_MV_21.xlsx   A+ PUMPS & SERVICE LLC   
2691882     2691882  163_Windham_MV_21.xlsx       AAA DURA GLAZE LLC   
2691883     2691883  163_Windham_MV_21.xlsx       AAA DURA GLAZE LLC   
2691884     2691884  163_Windham_MV_21.xlsx       AAA DURA GLAZE LLC   
2691885     2691885  163_Windham_MV_21.xlsx       AAA DURA GLAZE LLC   
2691886     2691886  163_Windham_MV_21.xlsx        ABBOTT CARLETON N   

                          street           city state   zip vehicle_year  \
2691877      114 PINE ORCHARD RD       BRANFORD    CT  6405         1974   
2691878             45 CHERRY ST  EAST HARTFORD    CT  6108         2008   
2691879  42 MACHINE SHOP HILL RD  SOUTH WINDHAM    CT  6266         2017   
2691880  42 MACHINE SHOP HILL RD  SOUTH WINDHAM    CT  6266         2020   
2691881  42 MACHINE SHOP HILL RD  SOUTH WINDHAM    CT  6266         2021   
2691882          292 SCOTLAND RD        WINDHAM    CT  6280         2021   
2691883          292 SCOTLAND RD        WINDHAM    CT  6280         2008   
2691884          292 SCOTLAND RD        WINDHAM    CT  6280         2008   
2691885          292 SCOTLAND RD        WINDHAM    CT  6280         2001   
2691886         143 BOULEVARD RD  NORTH WINDHAM    CT  6256         2008   

        vehicle_make vehicle_model vehicle_class         vehicle_id  \
2691877        FERRA       365GTB4            25              17993   
2691878         FORD      ECONOLIN             3  1FTSE34L88DA28181   
2691879         FORD      F350 SUP             2  1FT8W3DT7HEF28494   
2691880        BRAVO      SC712TA2            10  542BC1224LB031060   
2691881        REISE       TRAILER            10  55L1F2028MN008651   
2691882        VOLKS      ATLAS SE             1  1V2SR2CA8MC548989   
2691883         FORD          F150             3  1FTRF14V78KC77911   
2691884         FORD          F150             3  1FTRF12238KB63171   
2691885        DODGE        DAKOTA             3  1B7FL26X61S310631   
2691886        MITSU      GALANT E             1  4A3AB36F48E015544   

        lease_street  UID lease_city lease_state lease_zip  
2691877          NaN  NaN        NaN         NaN       NaN  
2691878          NaN  NaN        NaN         NaN       NaN  
2691879          NaN  NaN        NaN         NaN       NaN  
2691880          NaN  NaN        NaN         NaN       NaN  
2691881          NaN  NaN        NaN         NaN       NaN  
2691882          NaN  NaN        NaN         NaN       NaN  
2691883          NaN  NaN        NaN         NaN       NaN  
2691884          NaN  NaN        NaN         NaN       NaN  
2691885          NaN  NaN        NaN         NaN       NaN  
2691886          NaN  NaN        NaN         NaN       NaN

# Same checks

In [88]:
recompiled_data = pd.read_csv(raw_path / "2019-21_data_compiled_RN_092923.csv", chunksize = 1000)

In [89]:
missing_df_old = missing_df.copy(deep = True)

In [90]:
i = 0

missing_df = pd.DataFrame([])

for chunk in recompiled_data:
    
    # Give progress
    if (i % 1000 ==0):
        print(f"Currently on chunk number {i}")
        # display(df)
        
    # Create DF
    # Get the total number for that record_from
    total = chunk.groupby("record_from").count()["Unnamed: 0"].reset_index(name="count")
    
    # Get the number with a missing ZIP
    missing_zips = chunk.groupby("record_from")["zip"].apply(lambda x: x.isna().sum()).reset_index(name = "missing zip")
    
    # Get the number with a missing VIN
    missing_vins = chunk.groupby("record_from")["vehicle_id"].apply(lambda x: x.isna().sum()).reset_index(name = "missing VIN")
    
    # Missing address
    missing_streets = chunk.groupby("record_from")["street"].apply(lambda x: x.isna().sum()).reset_index(name = "missing street")
    
    # Get the number with state not in CT
    state_not_CT = chunk.groupby("record_from")["state"].apply(lambda x: (x!="CT").sum()).reset_index(name ="not CT")
    
    # Merge
    all_columns  = missing_zips.merge(missing_vins,
                                  how='outer',
                                  left_on = 'record_from',
                                  right_on = 'record_from')
    
    all_columns = all_columns.merge(missing_streets,
                                   how = 'outer',
                                   left_on = 'record_from',
                                   right_on = 'record_from')
    
    all_columns  = all_columns.merge(state_not_CT,
                                  how='outer',
                                  left_on = 'record_from',
                                  right_on = 'record_from')
    
    all_columns = all_columns.merge(total,
                                    how = 'outer',
                                    left_on = 'record_from',
                                    right_on = 'record_from')
    
    
    # Assign this to a DF to save
    missing_df = pd.concat([missing_df, all_columns])
    
    # add to count
    i +=1

missing_df = missing_df.groupby("record_from").sum()
missing_df = missing_df.reset_index()

# Get the details out of the from_record column
record_from_split = pd.DataFrame(list(missing_df["record_from"].str.split("_")),
                                 columns = ["number", "town", "suffix", "year", "other"])
record_from_split = record_from_split.drop("other", axis =1 )
record_from_split["year"] = record_from_split["year"].str.replace("\..*$", "")

Currently on chunk number 0


Unnamed: 0             record_from                     name  \
2691877     2691877  163_Windham_MV_21.xlsx  0 TO 60 MOTORSPORTS LLC   
2691878     2691878  163_Windham_MV_21.xlsx  A L FIRE PROTECTION LLC   
2691879     2691879  163_Windham_MV_21.xlsx   A+ PUMPS & SERVICE LLC   
2691880     2691880  163_Windham_MV_21.xlsx   A+ PUMPS & SERVICE LLC   
2691881     2691881  163_Windham_MV_21.xlsx   A+ PUMPS & SERVICE LLC   
2691882     2691882  163_Windham_MV_21.xlsx       AAA DURA GLAZE LLC   
2691883     2691883  163_Windham_MV_21.xlsx       AAA DURA GLAZE LLC   
2691884     2691884  163_Windham_MV_21.xlsx       AAA DURA GLAZE LLC   
2691885     2691885  163_Windham_MV_21.xlsx       AAA DURA GLAZE LLC   
2691886     2691886  163_Windham_MV_21.xlsx        ABBOTT CARLETON N   

                          street           city state   zip vehicle_year  \
2691877      114 PINE ORCHARD RD       BRANFORD    CT  6405         1974   
2691878             45 CHERRY ST  EAST HARTFORD    CT  6108         2008   
2691879  42 MACHINE SHOP HILL RD  SOUTH WINDHAM    CT  6266         2017   
2691880  42 MACHINE SHOP HILL RD  SOUTH WINDHAM    CT  6266         2020   
2691881  42 MACHINE SHOP HILL RD  SOUTH WINDHAM    CT  6266         2021   
2691882          292 SCOTLAND RD        WINDHAM    CT  6280         2021   
2691883          292 SCOTLAND RD        WINDHAM    CT  6280         2008   
2691884          292 SCOTLAND RD        WINDHAM    CT  6280         2008   
2691885          292 SCOTLAND RD        WINDHAM    CT  6280         2001   
2691886         143 BOULEVARD RD  NORTH WINDHAM    CT  6256         2008   

        vehicle_make vehicle_model vehicle_class         vehicle_id  \
2691877        FERRA       365GTB4            25              17993   
2691878         FORD      ECONOLIN             3  1FTSE34L88DA28181   
2691879         FORD      F350 SUP             2  1FT8W3DT7HEF28494   
2691880        BRAVO      SC712TA2            10  542BC1224LB031060   
2691881        REISE       TRAILER            10  55L1F2028MN008651   
2691882        VOLKS      ATLAS SE             1  1V2SR2CA8MC548989   
2691883         FORD          F150             3  1FTRF14V78KC77911   
2691884         FORD          F150             3  1FTRF12238KB63171   
2691885        DODGE        DAKOTA             3  1B7FL26X61S310631   
2691886        MITSU      GALANT E             1  4A3AB36F48E015544   

        lease_street  UID lease_city lease_state lease_zip  
2691877          NaN  NaN        NaN         NaN       NaN  
2691878          NaN  NaN        NaN         NaN       NaN  
2691879          NaN  NaN        NaN         NaN       NaN  
2691880          NaN  NaN        NaN         NaN       NaN  
2691881          NaN  NaN        NaN         NaN       NaN  
2691882          NaN  NaN        NaN         NaN       NaN  
2691883          NaN  NaN        NaN         NaN       NaN  
2691884          NaN  NaN        NaN         NaN       NaN  
2691885          NaN  NaN        NaN         NaN       NaN  
2691886          NaN  NaN        NaN         NaN       NaN

Currently on chunk number 1000


Unnamed: 0             record_from                     name  \
2691877     2691877  163_Windham_MV_21.xlsx  0 TO 60 MOTORSPORTS LLC   
2691878     2691878  163_Windham_MV_21.xlsx  A L FIRE PROTECTION LLC   
2691879     2691879  163_Windham_MV_21.xlsx   A+ PUMPS & SERVICE LLC   
2691880     2691880  163_Windham_MV_21.xlsx   A+ PUMPS & SERVICE LLC   
2691881     2691881  163_Windham_MV_21.xlsx   A+ PUMPS & SERVICE LLC   
2691882     2691882  163_Windham_MV_21.xlsx       AAA DURA GLAZE LLC   
2691883     2691883  163_Windham_MV_21.xlsx       AAA DURA GLAZE LLC   
2691884     2691884  163_Windham_MV_21.xlsx       AAA DURA GLAZE LLC   
2691885     2691885  163_Windham_MV_21.xlsx       AAA DURA GLAZE LLC   
2691886     2691886  163_Windham_MV_21.xlsx        ABBOTT CARLETON N   

                          street           city state   zip vehicle_year  \
2691877      114 PINE ORCHARD RD       BRANFORD    CT  6405         1974   
2691878             45 CHERRY ST  EAST HARTFORD    CT  6108         2008   
2691879  42 MACHINE SHOP HILL RD  SOUTH WINDHAM    CT  6266         2017   
2691880  42 MACHINE SHOP HILL RD  SOUTH WINDHAM    CT  6266         2020   
2691881  42 MACHINE SHOP HILL RD  SOUTH WINDHAM    CT  6266         2021   
2691882          292 SCOTLAND RD        WINDHAM    CT  6280         2021   
2691883          292 SCOTLAND RD        WINDHAM    CT  6280         2008   
2691884          292 SCOTLAND RD        WINDHAM    CT  6280         2008   
2691885          292 SCOTLAND RD        WINDHAM    CT  6280         2001   
2691886         143 BOULEVARD RD  NORTH WINDHAM    CT  6256         2008   

        vehicle_make vehicle_model vehicle_class         vehicle_id  \
2691877        FERRA       365GTB4            25              17993   
2691878         FORD      ECONOLIN             3  1FTSE34L88DA28181   
2691879         FORD      F350 SUP             2  1FT8W3DT7HEF28494   
2691880        BRAVO      SC712TA2            10  542BC1224LB031060   
2691881        REISE       TRAILER            10  55L1F2028MN008651   
2691882        VOLKS      ATLAS SE             1  1V2SR2CA8MC548989   
2691883         FORD          F150             3  1FTRF14V78KC77911   
2691884         FORD          F150             3  1FTRF12238KB63171   
2691885        DODGE        DAKOTA             3  1B7FL26X61S310631   
2691886        MITSU      GALANT E             1  4A3AB36F48E015544   

        lease_street  UID lease_city lease_state lease_zip  
2691877          NaN  NaN        NaN         NaN       NaN  
2691878          NaN  NaN        NaN         NaN       NaN  
2691879          NaN  NaN        NaN         NaN       NaN  
2691880          NaN  NaN        NaN         NaN       NaN  
2691881          NaN  NaN        NaN         NaN       NaN  
2691882          NaN  NaN        NaN         NaN       NaN  
2691883          NaN  NaN        NaN         NaN       NaN  
2691884          NaN  NaN        NaN         NaN       NaN  
2691885          NaN  NaN        NaN         NaN       NaN  
2691886          NaN  NaN        NaN         NaN       NaN

Currently on chunk number 2000


Unnamed: 0             record_from                     name  \
2691877     2691877  163_Windham_MV_21.xlsx  0 TO 60 MOTORSPORTS LLC   
2691878     2691878  163_Windham_MV_21.xlsx  A L FIRE PROTECTION LLC   
2691879     2691879  163_Windham_MV_21.xlsx   A+ PUMPS & SERVICE LLC   
2691880     2691880  163_Windham_MV_21.xlsx   A+ PUMPS & SERVICE LLC   
2691881     2691881  163_Windham_MV_21.xlsx   A+ PUMPS & SERVICE LLC   
2691882     2691882  163_Windham_MV_21.xlsx       AAA DURA GLAZE LLC   
2691883     2691883  163_Windham_MV_21.xlsx       AAA DURA GLAZE LLC   
2691884     2691884  163_Windham_MV_21.xlsx       AAA DURA GLAZE LLC   
2691885     2691885  163_Windham_MV_21.xlsx       AAA DURA GLAZE LLC   
2691886     2691886  163_Windham_MV_21.xlsx        ABBOTT CARLETON N   

                          street           city state   zip vehicle_year  \
2691877      114 PINE ORCHARD RD       BRANFORD    CT  6405         1974   
2691878             45 CHERRY ST  EAST HARTFORD    CT  6108         2008   
2691879  42 MACHINE SHOP HILL RD  SOUTH WINDHAM    CT  6266         2017   
2691880  42 MACHINE SHOP HILL RD  SOUTH WINDHAM    CT  6266         2020   
2691881  42 MACHINE SHOP HILL RD  SOUTH WINDHAM    CT  6266         2021   
2691882          292 SCOTLAND RD        WINDHAM    CT  6280         2021   
2691883          292 SCOTLAND RD        WINDHAM    CT  6280         2008   
2691884          292 SCOTLAND RD        WINDHAM    CT  6280         2008   
2691885          292 SCOTLAND RD        WINDHAM    CT  6280         2001   
2691886         143 BOULEVARD RD  NORTH WINDHAM    CT  6256         2008   

        vehicle_make vehicle_model vehicle_class         vehicle_id  \
2691877        FERRA       365GTB4            25              17993   
2691878         FORD      ECONOLIN             3  1FTSE34L88DA28181   
2691879         FORD      F350 SUP             2  1FT8W3DT7HEF28494   
2691880        BRAVO      SC712TA2            10  542BC1224LB031060   
2691881        REISE       TRAILER            10  55L1F2028MN008651   
2691882        VOLKS      ATLAS SE             1  1V2SR2CA8MC548989   
2691883         FORD          F150             3  1FTRF14V78KC77911   
2691884         FORD          F150             3  1FTRF12238KB63171   
2691885        DODGE        DAKOTA             3  1B7FL26X61S310631   
2691886        MITSU      GALANT E             1  4A3AB36F48E015544   

        lease_street  UID lease_city lease_state lease_zip  
2691877          NaN  NaN        NaN         NaN       NaN  
2691878          NaN  NaN        NaN         NaN       NaN  
2691879          NaN  NaN        NaN         NaN       NaN  
2691880          NaN  NaN        NaN         NaN       NaN  
2691881          NaN  NaN        NaN         NaN       NaN  
2691882          NaN  NaN        NaN         NaN       NaN  
2691883          NaN  NaN        NaN         NaN       NaN  
2691884          NaN  NaN        NaN         NaN       NaN  
2691885          NaN  NaN        NaN         NaN       NaN  
2691886          NaN  NaN        NaN         NaN       NaN

Currently on chunk number 3000


Unnamed: 0             record_from                     name  \
2691877     2691877  163_Windham_MV_21.xlsx  0 TO 60 MOTORSPORTS LLC   
2691878     2691878  163_Windham_MV_21.xlsx  A L FIRE PROTECTION LLC   
2691879     2691879  163_Windham_MV_21.xlsx   A+ PUMPS & SERVICE LLC   
2691880     2691880  163_Windham_MV_21.xlsx   A+ PUMPS & SERVICE LLC   
2691881     2691881  163_Windham_MV_21.xlsx   A+ PUMPS & SERVICE LLC   
2691882     2691882  163_Windham_MV_21.xlsx       AAA DURA GLAZE LLC   
2691883     2691883  163_Windham_MV_21.xlsx       AAA DURA GLAZE LLC   
2691884     2691884  163_Windham_MV_21.xlsx       AAA DURA GLAZE LLC   
2691885     2691885  163_Windham_MV_21.xlsx       AAA DURA GLAZE LLC   
2691886     2691886  163_Windham_MV_21.xlsx        ABBOTT CARLETON N   

                          street           city state   zip vehicle_year  \
2691877      114 PINE ORCHARD RD       BRANFORD    CT  6405         1974   
2691878             45 CHERRY ST  EAST HARTFORD    CT  6108         2008   
2691879  42 MACHINE SHOP HILL RD  SOUTH WINDHAM    CT  6266         2017   
2691880  42 MACHINE SHOP HILL RD  SOUTH WINDHAM    CT  6266         2020   
2691881  42 MACHINE SHOP HILL RD  SOUTH WINDHAM    CT  6266         2021   
2691882          292 SCOTLAND RD        WINDHAM    CT  6280         2021   
2691883          292 SCOTLAND RD        WINDHAM    CT  6280         2008   
2691884          292 SCOTLAND RD        WINDHAM    CT  6280         2008   
2691885          292 SCOTLAND RD        WINDHAM    CT  6280         2001   
2691886         143 BOULEVARD RD  NORTH WINDHAM    CT  6256         2008   

        vehicle_make vehicle_model vehicle_class         vehicle_id  \
2691877        FERRA       365GTB4            25              17993   
2691878         FORD      ECONOLIN             3  1FTSE34L88DA28181   
2691879         FORD      F350 SUP             2  1FT8W3DT7HEF28494   
2691880        BRAVO      SC712TA2            10  542BC1224LB031060   
2691881        REISE       TRAILER            10  55L1F2028MN008651   
2691882        VOLKS      ATLAS SE             1  1V2SR2CA8MC548989   
2691883         FORD          F150             3  1FTRF14V78KC77911   
2691884         FORD          F150             3  1FTRF12238KB63171   
2691885        DODGE        DAKOTA             3  1B7FL26X61S310631   
2691886        MITSU      GALANT E             1  4A3AB36F48E015544   

        lease_street  UID lease_city lease_state lease_zip  
2691877          NaN  NaN        NaN         NaN       NaN  
2691878          NaN  NaN        NaN         NaN       NaN  
2691879          NaN  NaN        NaN         NaN       NaN  
2691880          NaN  NaN        NaN         NaN       NaN  
2691881          NaN  NaN        NaN         NaN       NaN  
2691882          NaN  NaN        NaN         NaN       NaN  
2691883          NaN  NaN        NaN         NaN       NaN  
2691884          NaN  NaN        NaN         NaN       NaN  
2691885          NaN  NaN        NaN         NaN       NaN  
2691886          NaN  NaN        NaN         NaN       NaN

Currently on chunk number 4000


Unnamed: 0             record_from                     name  \
2691877     2691877  163_Windham_MV_21.xlsx  0 TO 60 MOTORSPORTS LLC   
2691878     2691878  163_Windham_MV_21.xlsx  A L FIRE PROTECTION LLC   
2691879     2691879  163_Windham_MV_21.xlsx   A+ PUMPS & SERVICE LLC   
2691880     2691880  163_Windham_MV_21.xlsx   A+ PUMPS & SERVICE LLC   
2691881     2691881  163_Windham_MV_21.xlsx   A+ PUMPS & SERVICE LLC   
2691882     2691882  163_Windham_MV_21.xlsx       AAA DURA GLAZE LLC   
2691883     2691883  163_Windham_MV_21.xlsx       AAA DURA GLAZE LLC   
2691884     2691884  163_Windham_MV_21.xlsx       AAA DURA GLAZE LLC   
2691885     2691885  163_Windham_MV_21.xlsx       AAA DURA GLAZE LLC   
2691886     2691886  163_Windham_MV_21.xlsx        ABBOTT CARLETON N   

                          street           city state   zip vehicle_year  \
2691877      114 PINE ORCHARD RD       BRANFORD    CT  6405         1974   
2691878             45 CHERRY ST  EAST HARTFORD    CT  6108         2008   
2691879  42 MACHINE SHOP HILL RD  SOUTH WINDHAM    CT  6266         2017   
2691880  42 MACHINE SHOP HILL RD  SOUTH WINDHAM    CT  6266         2020   
2691881  42 MACHINE SHOP HILL RD  SOUTH WINDHAM    CT  6266         2021   
2691882          292 SCOTLAND RD        WINDHAM    CT  6280         2021   
2691883          292 SCOTLAND RD        WINDHAM    CT  6280         2008   
2691884          292 SCOTLAND RD        WINDHAM    CT  6280         2008   
2691885          292 SCOTLAND RD        WINDHAM    CT  6280         2001   
2691886         143 BOULEVARD RD  NORTH WINDHAM    CT  6256         2008   

        vehicle_make vehicle_model vehicle_class         vehicle_id  \
2691877        FERRA       365GTB4            25              17993   
2691878         FORD      ECONOLIN             3  1FTSE34L88DA28181   
2691879         FORD      F350 SUP             2  1FT8W3DT7HEF28494   
2691880        BRAVO      SC712TA2            10  542BC1224LB031060   
2691881        REISE       TRAILER            10  55L1F2028MN008651   
2691882        VOLKS      ATLAS SE             1  1V2SR2CA8MC548989   
2691883         FORD          F150             3  1FTRF14V78KC77911   
2691884         FORD          F150             3  1FTRF12238KB63171   
2691885        DODGE        DAKOTA             3  1B7FL26X61S310631   
2691886        MITSU      GALANT E             1  4A3AB36F48E015544   

        lease_street  UID lease_city lease_state lease_zip  
2691877          NaN  NaN        NaN         NaN       NaN  
2691878          NaN  NaN        NaN         NaN       NaN  
2691879          NaN  NaN        NaN         NaN       NaN  
2691880          NaN  NaN        NaN         NaN       NaN  
2691881          NaN  NaN        NaN         NaN       NaN  
2691882          NaN  NaN        NaN         NaN       NaN  
2691883          NaN  NaN        NaN         NaN       NaN  
2691884          NaN  NaN        NaN         NaN       NaN  
2691885          NaN  NaN        NaN         NaN       NaN  
2691886          NaN  NaN        NaN         NaN       NaN

Currently on chunk number 5000


Unnamed: 0             record_from                     name  \
2691877     2691877  163_Windham_MV_21.xlsx  0 TO 60 MOTORSPORTS LLC   
2691878     2691878  163_Windham_MV_21.xlsx  A L FIRE PROTECTION LLC   
2691879     2691879  163_Windham_MV_21.xlsx   A+ PUMPS & SERVICE LLC   
2691880     2691880  163_Windham_MV_21.xlsx   A+ PUMPS & SERVICE LLC   
2691881     2691881  163_Windham_MV_21.xlsx   A+ PUMPS & SERVICE LLC   
2691882     2691882  163_Windham_MV_21.xlsx       AAA DURA GLAZE LLC   
2691883     2691883  163_Windham_MV_21.xlsx       AAA DURA GLAZE LLC   
2691884     2691884  163_Windham_MV_21.xlsx       AAA DURA GLAZE LLC   
2691885     2691885  163_Windham_MV_21.xlsx       AAA DURA GLAZE LLC   
2691886     2691886  163_Windham_MV_21.xlsx        ABBOTT CARLETON N   

                          street           city state   zip vehicle_year  \
2691877      114 PINE ORCHARD RD       BRANFORD    CT  6405         1974   
2691878             45 CHERRY ST  EAST HARTFORD    CT  6108         2008   
2691879  42 MACHINE SHOP HILL RD  SOUTH WINDHAM    CT  6266         2017   
2691880  42 MACHINE SHOP HILL RD  SOUTH WINDHAM    CT  6266         2020   
2691881  42 MACHINE SHOP HILL RD  SOUTH WINDHAM    CT  6266         2021   
2691882          292 SCOTLAND RD        WINDHAM    CT  6280         2021   
2691883          292 SCOTLAND RD        WINDHAM    CT  6280         2008   
2691884          292 SCOTLAND RD        WINDHAM    CT  6280         2008   
2691885          292 SCOTLAND RD        WINDHAM    CT  6280         2001   
2691886         143 BOULEVARD RD  NORTH WINDHAM    CT  6256         2008   

        vehicle_make vehicle_model vehicle_class         vehicle_id  \
2691877        FERRA       365GTB4            25              17993   
2691878         FORD      ECONOLIN             3  1FTSE34L88DA28181   
2691879         FORD      F350 SUP             2  1FT8W3DT7HEF28494   
2691880        BRAVO      SC712TA2            10  542BC1224LB031060   
2691881        REISE       TRAILER            10  55L1F2028MN008651   
2691882        VOLKS      ATLAS SE             1  1V2SR2CA8MC548989   
2691883         FORD          F150             3  1FTRF14V78KC77911   
2691884         FORD          F150             3  1FTRF12238KB63171   
2691885        DODGE        DAKOTA             3  1B7FL26X61S310631   
2691886        MITSU      GALANT E             1  4A3AB36F48E015544   

        lease_street  UID lease_city lease_state lease_zip  
2691877          NaN  NaN        NaN         NaN       NaN  
2691878          NaN  NaN        NaN         NaN       NaN  
2691879          NaN  NaN        NaN         NaN       NaN  
2691880          NaN  NaN        NaN         NaN       NaN  
2691881          NaN  NaN        NaN         NaN       NaN  
2691882          NaN  NaN        NaN         NaN       NaN  
2691883          NaN  NaN        NaN         NaN       NaN  
2691884          NaN  NaN        NaN         NaN       NaN  
2691885          NaN  NaN        NaN         NaN       NaN  
2691886          NaN  NaN        NaN         NaN       NaN

C:\Users\rajan\AppData\Local\Temp\ipykernel_21188\2552312415.py:63: FutureWarning: The default value of regex will change from True to False in a future version.
  record_from_split["year"] = record_from_split["year"].str.replace("\..*$", "")


**Consider missing VINs**

In [87]:
 missing_df.sort_values("missing VIN", ascending = False).head(20)

record_from  missing zip  missing VIN  not CT  \
133      11_Bloomfield_MVData_2019.csv            0        19643    1630   
107         107_Orange_MVData_2020.csv            0        14650    1939   
143              124_Seymour_MV_21.csv            0        13860    1119   
14   016_Bridgewater_MV_21_ALTERED.csv         4886         4886    5074   
179            141_Thompson_MV_21.xlsx            0          457     447   
136          121_Salem_MVData_2019.csv            0          218     126   
96          101_North_Haven_MV_21.xlsx           84           84    2978   
76           083_Middletown_MV_21.xlsx           82           82    3060   
167       136_Sterling_MVData_2019.csv           71           69     223   
103       105_Old_Lyme_MVData_2019.csv           37           37     530   
223             163_Windham_MV_21.xlsx           37           37     848   
168            136_Sterling_MV_21.xlsx           36           34     161   
222        163_Windham_MVData_2020.csv           34           34     864   
190      147_Voluntown_MVData_2020.csv           29           29     105   
22              024_Chaplin_MV_21.xlsx           29           29     104   
24    028_Colchester_MV_21_ALTERED.csv            8            8     701   
82          089_New_Britain_MV_21.xlsx            2            2    2174   
287         63_Hampton_MVData_2019.csv            2            2      67   
70    077_Manchester_MV_21_ALTERED.csv            3            2    2974   
286          62_Hamden_MVData_2020.csv            2            2    3796   

     count  
133  19643  
107  14650  
143  15410  
14    7254  
179  10054  
136   4914  
96   24425  
76   35302  
167   4249  
103   9135  
223  16516  
168   4091  
222  16228  
190   3129  
22    2645  
24   16219  
82   45226  
287   2446  
70   44487  
286  38900

**MISSING VINs**
* Bloomfield file for 2019 does not contain VINs
* Orange file for 2019 does not contain VINs
* Seymour file has major issues for missing VINs
* Bridgewater file - has this many missing VINs in the underlying file.
* Thompson file - has 457 missing VINs in the underlying file anyway

**Consider missing street addresses**

In [91]:
 missing_df.sort_values("missing street", ascending = False).head(20)

record_from  missing zip  missing VIN  \
109              107_Orange_MV_21.xlsx        13522            0   
248         27_Clinton_MVData_2020.csv        13202            0   
33                 037_Derby_MV_21.xls         9511            0   
15   016_Bridgewater_MV_21_ALTERED.csv         4886         4886   
114       10_Bethlehem_MVData_2019.csv            0            0   
119        112_Pomfret_MVData_2020.csv         4130            0   
49               055_Goshen_MV_21.XLSX         4004            0   
92              098_Norfolk_MV_21.xlsx         1945            0   
97          101_North_Haven_MV_21.xlsx           84           84   
77           083_Middletown_MV_21.xlsx           82           82   
168       136_Sterling_MVData_2019.csv           71           69   
104       105_Old_Lyme_MVData_2019.csv           37           37   
224             163_Windham_MV_21.xlsx           37           37   
169            136_Sterling_MV_21.xlsx           36           34   
223        163_Windham_MVData_2020.csv           34           34   
23              024_Chaplin_MV_21.xlsx           29           29   
191      147_Voluntown_MVData_2020.csv           29           29   
25    028_Colchester_MV_21_ALTERED.csv            8            8   
180            141_Thompson_MV_21.xlsx            0          457   
71    077_Manchester_MV_21_ALTERED.csv            3            2   

     missing street  not CT  count  
109           13522   13522  13522  
248           13202   13202  13202  
33             9511    9511   9511  
15             4886    5074   7254  
114            4760     206   4760  
119            4130    4130   4130  
49             4004    4004   4004  
92             1945    1945   1945  
97               84    2978  24425  
77               82    3060  35302  
168              69     223   4249  
104              37     530   9135  
224              37     848  16516  
169              34     161   4091  
223              34     864  16228  
23               29     104   2645  
191              29     105   3129  
25                8     701  16219  
180               7     447  10054  
71                3    2974  44487

Orange 2021, Clinton 2020, Derby 2021, Bridgewater 2021 - All missing this info anyway.
Bethlehem 2019, Pomfret 2020, Goshen 2021 - all the same
Norfolk 2019 - is missing 1945 addresses in the underlying file

**Consider missing ZIP codes**

In [92]:
missing_df.sort_values("missing zip", ascending = False).head(20)

record_from  missing zip  missing VIN  \
323           96_Newington_MVData_2020.csv        27612            0   
39                 045_East_Lyme_MV_21.csv        17616            0   
3                      004_Avon_MV_21.xlsx        15860            0   
273                 4_Avon_MVData_2020.csv        15654            0   
66                  072_Ledyard_MV_21.xlsx        13858            0   
109                  107_Orange_MV_21.xlsx        13522            0   
248             27_Clinton_MVData_2020.csv        13202            0   
160             131_Southington_MV_21.xlsx        10089            0   
33                     037_Derby_MV_21.xls         9511            0   
129                 117_Redding_MV_21.xlsx         8171            0   
243          22_Canterbury_MVData_2019.csv         6334            0   
15       016_Bridgewater_MV_21_ALTERED.csv         4886         4886   
267           44_East_Lyme_MVData_2019.csv         4301            0   
119            112_Pomfret_MVData_2020.csv         4130            0   
49                   055_Goshen_MV_21.XLSX         4004            0   
150                 127_Sherman_MV_21.xlsx         3822            0   
149            127_Sherman_MVData_2020.csv         3748            0   
92                  098_Norfolk_MV_21.xlsx         1945            0   
27                 031_Cornwall_MV_21.xlsx         1785            0   
303  77_Manchester_MVData_2019_ALTERED.csv          221            0   

     missing street  not CT  count  
323               0   27612  27612  
39                0   17616  17616  
3                 0    1547  15860  
273               0   15654  15654  
66                0   13858  13858  
109           13522   13522  13522  
248           13202   13202  13202  
160               2   12764  42996  
33             9511    9511   9511  
129               0    8171   8171  
243               0    6334   6334  
15             4886    5074   7254  
267               0    5215  17448  
119            4130    4130   4130  
49             4004    4004   4004  
150               0    3822   3822  
149               0    3748   3748  
92             1945    1945   1945  
27                0     142   1785  
303               0    3456  45785

* Newington - **error** - the column should be "ZIP"
* East Lyme - **partial error** - the Zip code data is under "Address" cont for SOME of them
* Avon 2021 - **error** - the Zip code data is under ZIP
* Avon 2020 - **error** - the ZIP should be under zip1
* Ledyard 2021 - **error** - The Zip is under ADDRESS but must be split first
* Orange 2021 - **not an error** - the ZIP is genuinely missing in the underlying file
* Clinton - is genuinely missing
* Southington - 10,089 ZIPs are genuinely missing
* Derby - is genuinely missing
* Redding - is genuinely missing in the underlying file
* Canterbury 2019 - **error** - the ZIP information is under "ZIP"
* Bridgewater 2021 - real
* East Lyme 2019 - is under Zip1 and Zip2
* Pomfret - Real
* Goshen - Real
* Sherman 2021 - Error - is under zip
* Sherman 2020 - Error - is under ZIP
* Norfolk 2021 - genuinely missing
* Cornwall 2021 - **error** - under ZIP
* Manchester 2019 - **error** - under Unnamed 6 - Unnamed 8